listcrunch/listcrunch.py

```diff
@@ -1,12 +1,13 @@
+from __future__ import unicode_literals
+
 import collections
 
 class ListCrunchError(Exception):
     pass
 
 def crunch_collection(collection):
     parts = []
-    for value in collection:
-        nums = sorted(collection[value])
+    for value, nums in sorted(collection.items(), key=lambda x: x[1]):
```

Comment:
You rely on the implementation detail of python3 that dictionaries arr ordered - so I sort by the `nums` here to ensure that order, as specified in your tests

You also sorted the `nums`, which I removed, as those should always be sorted by construction.  If we consider `crunch_collection` part of the public API (which I guess it kind of is, since it is not `_crunch_collection`, but I don't know that it should be, as I think we really just want to call `crunch` and `uncrunch`), then it might be needed